.github/scripts/test_and_publish.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+import subprocess, sys, os, tempfile
+from distantrs import Invocation
+from glob import glob
+
+INVOCATION_DATA_PATH = '/tmp/distant-rs-invocation.txt'
+
+# required to find listener.py
+this_path = os.path.abspath(os.path.dirname(__file__))
+test_script_path = f"{os.environ.get('GITHUB_WORKSPACE')}/renode/test.sh"
+inv_data_str = ""
+robot_name = glob("*.robot")[0].split(".")[0]
+
+with open(INVOCATION_DATA_PATH, 'r') as f:
+    inv_data_str = f.read()
+
+inv_data_lst = inv_data_str.split("--")
+
+i = Invocation(
+        invocation_id=inv_data_lst[0],
+        auth_token=inv_data_lst[1]
+        )
+
+i.announce_target(robot_name)
+
+tmp = tempfile.NamedTemporaryFile()
+with open(tmp.name, 'w') as log:
+    # run the actual testing procedure
+    subprocess_args = [
+            test_script_path, 
+            "--listener",
+            os.path.join(this_path, f"results_listener.py:{inv_data_str}")
+                ]
```

Comment:
something wrong with formatting here

Reply:
That's how my vim does it. I'll fix it though.